### PR TITLE
Adding loader for TAU Urban Acoustic Scenes 2019 dataset

### LINKED ISSUE #38: Adding TAU Urban Acoustic Scenes 2019 loader
I'm working on a loader for TAU Urban Acoustic Scenes 2019 ([development](https://zenodo.org/record/2589280), [leaderboard](https://zenodo.org/record/2672993) end [evaluation](https://zenodo.org/record/3063822) subsets). This dataset was used in the [Task 1 - subtask A](http://dcase.community/challenge2019/task-acoustic-scene-classification#description) of the DCASE challenge 2019.

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

#### TODOs
- [x] Complete Clip.to_jams() function.
- [x] Add test_to_jams() function.
- [x] Complete documentation.

### CODE REVIEW

File: soundata/datasets/tau2019uas.py
Comment (magdalenafuentes) [inline soundata/datasets/tau2019uas.py:224]: super complete ❤️

File: soundata/datasets/tau2019uas.py
Comment (magdalenafuentes) [inline soundata/datasets/tau2019uas.py]: remove

File: soundata/datasets/tau2019uas.py
Comment (magdalenafuentes) [inline soundata/datasets/tau2019uas.py:534]: This line is no covered by the tests

File: soundata/datasets/tau2019uas.py
Comment (magdalenafuentes) [inline soundata/datasets/tau2019uas.py:543]: missing in readme of `clip`

File: soundata/datasets/tau2019uas.py
Comment (magdalenafuentes) [inline soundata/datasets/tau2019uas.py]: I'm a bit confused about the repetition of code in `_metadata`, can you merge (if there are differences) this block with the one below?

File: tests/datasets/test_tau2019uas.py
Comment (magdalenafuentes) [inline tests/datasets/test_tau2019uas.py:44]: ❤️

File: tests/resources/sound_datasets/tau2019uas/TAU-urban-acoustic-scenes-2019-development/evaluation_setup/fold1_evaluate.csv
Comment (magdalenafuentes) [inline tests/resources/sound_datasets/tau2019uas/TAU-urban-acoustic-scenes-2019-development/evaluation_setup/fold1_evaluate.csv:1]: ✂️

File: tests/resources/sound_datasets/tau2019uas/TAU-urban-acoustic-scenes-2019-development/evaluation_setup/fold1_test.csv
Comment (magdalenafuentes) [inline tests/resources/sound_datasets/tau2019uas/TAU-urban-acoustic-scenes-2019-development/evaluation_setup/fold1_test.csv:1]: ✂️ and same for those below

**Review by magdalenafuentes — APPROVED** [review]
@pzinemanas looks great! just left minor comments and good to go! Feel free to merge after those minor changes (and let me know if something is not clear)

File: soundata/datasets/tau2019uas.py
Comment (pzinemanas) [reply soundata/datasets/tau2019uas.py]: The problem here is that some information is in the development metadata (e.g the identifier), some is in the evaluation setup files (e.g. split), and some is in both places (e.g. city, scene_label). I've decided to use metadata file for everything (first round) and update the split looking at the evaluation setup files (second round). Besides, in the second round, I have to define the metadata_index for evaluation and leaderboard datasets (they don't have metadata file). @magdalenafuentes  do you find a better way to do it?

File: soundata/datasets/tau2019uas.py
Comment (pzinemanas) [reply soundata/datasets/tau2019uas.py:534]: Thanks @magdalenafuentes ! Thanks to this, I've found a bug. Evaluation and leaderboard datasets share filename, and therefore the filename is not a unique `clip_id`. To solve this, I've changed the `clip_id = {subset}/{filename}`. This sounds good to you @magdalenafuentes ?

Comment (magdalenafuentes) [reply soundata/datasets/tau2019uas.py:534]: Sounds good to me, so you would have something like `eval/00123`? could maybe be `eval_00123` or `eval00123` so we have same name conventions across `soundata`

**Review by magdalenafuentes — APPROVED** [review]

File: soundata/datasets/tau2019uas.py
Comment (magdalenafuentes) [reply soundata/datasets/tau2019uas.py]: mmmm... ok I see. I was also a bit worried that you might be overwriting the metadata_index dictionary if you have repeated `clip_id`, but I guess this would be completely avoided with your new naming convention. I'm good leaving this like it is!

File: soundata/datasets/tau2019uas.py
Comment (pzinemanas) [reply soundata/datasets/tau2019uas.py:534]: I'm using `eval/00123` so it's easy to split it with `clip_id.split('/')`. Note that `_` is often used in file names.

Comment (magdalenafuentes) [reply soundata/datasets/tau2019uas.py:534]: fair enough! 👍

File: soundata/datasets/tau2019uas.py
Comment (pzinemanas) [reply soundata/datasets/tau2019uas.py]: Yes, actually I was overwriting it, but as you said, now should be fixed.

**Comment by justinsalamon** [comment]
@pzinemanas nice work! You'll need to update the branch from master before merging (can be done via the PR UI). 

@magdalenafuentes once this PR is merged we should open a quick PR to  bump the version up. We should probably do this every time a new dataset loader is merged into master. We should also update the main README (list of datasets).

**Comment by magdalenafuentes** [comment]
About the `README`, my experience with `mirdata` is that is better to remove the list from there, since it requires updating one more thing every time we merge a loader, and it gets ugly as the library goes.

About the version, we should discuss briefly how we wanna do it. I think one possibility is to do a pip `non stable` release every time a new loader gets in, and then after they've been tried a bit move to stable. But I'm not sure yet what's the best. Let's move that discussion somewhere else

**Pull request merged**
